clippy_lints/src/functions/mod.rs

```diff
@@ -399,6 +400,26 @@ declare_clippy_lint! {
     "renamed function parameters in trait implementation"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Warns when a function signature uses `&Option<T>` instead of `Option<&T>`.
+    /// ### Why is this bad?
+    /// More flexibility, better memory optimization, and more idiomatic Rust code.
+    /// See [YouTube video](https://www.youtube.com/watch?v=6c7pZYP_iIE)
```

Comment:
```suggestion
    /// See this [YouTube video](https://www.youtube.com/watch?v=6c7pZYP_iIE)
```

Reply:
I believe that adding the video title and author/channel would be nice, as it would serve as `alt` text

Reply:
@Rudxain how can you add an alt text to a non-image/video link? I thought the `![Alt text](...)` is only for images and such. Please give an example of what you think this markdown should be like. I couldn't find any other code in the rust repo that had alt text like this. Thx!

Reply:
If possible, you could try to summarize succintly the key reasoning from the YouTube video for why this might enable better optimizations.

Reply:
thanks, updated.

Reply:
You should also copy-pasta your summary to your PR description, makes it easier to understand why your PR included these changes 😸

Reply:
> add an alt text to a non-image/video link?

I'm sorry for not being clear. I meant it as "equivalent to `alt`" not literally an `alt`. It doesn't matter how you add the extra information, what matters is the information itself.

I was thinking of a good ol' link: the "name" of the link can be used as an `alt`:
```md
[This link points to an example web page](https://example.org)
```